src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzIamStep.java

```diff
@@ -58,6 +59,9 @@ public StepResult undoStep(FlightContext context) {
             // suppress exception
             logger.error("NEEDS CLEANUP: delete sam resource for snapshot " + snapshotId.toString());
             logger.warn(ex.getMessage());
+        } catch (IamNotFoundException ex) {
```

Comment:
Elsewhere this case is handled using `NotFoundException` not `IamNotFoundException`. https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotAuthzResource.java#L30

Reply:
Hmm...not sure if it's better to be more or less specific but I agree it should be consistent.  I'll update to match snapshot delete

Reply:
Wouldn't it be sufficient to test that the undostep no longer throws in the case where a snapshot no longer exists? This could be done with a mocked `IamService`.

Reply:
I think this was meant as a reply to the question of unit vs. integration.  I really wanted to test to make sure that the end to end behavior worked as expected (e.g. that we're correctly catching and then correctly backing out)